k9mail/src/main/java/com/fsck/k9/service/MailService.java

```diff
@@ -342,7 +344,10 @@ private void reschedulePushers(boolean hasConnectivity, boolean doBackground) {
 
     private void setupPushers() {
         boolean pushing = false;
-        for (Account account : Preferences.getPreferences(MailService.this).getAccounts()) {
+        List<Account> accounts =
+                new ArrayList<Account>(Preferences.getPreferences(MailService.this)
```

Comment:
`Collections.unmodifiableList()` only wraps the underlying list and prevents modifying the list when using the wrapped object. Changes to the underlying list are still reflected in the unmodifiable list.